specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+IFrame Memory Usage API
+===
+
+# Background
+The WebView2 team has been asked to provide support for monitoring iframe 
+memory usage. The end developer needs to know how much memory each iframe 
+consumes. To do that, they need to get the renderer process ID for each 
+iframe. However, WebView2 only tracks the render frame host for top-level iframes. 
+This limits the [ICoreWebView2Frame2](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2frame?view=webview2-1.0.1518.46)
+APIs for first-level iframes only. For instance, we raise the FrameCreated 
+event for first-level iframes, but not for nested iframes with depth greater than 1. 
+Also, WebView2 lacks an API to get the renderer process ID for iframes.
+<pre>
+                           main_frame  [depth:0]

```

Comment:
This diagram implies that the WebView2 is itself a frame. I thought we reserved the term "frame" to refer to IFrames inside the top-level document, not for the top-level document itself. (You can't QI ICoreWebView2 for ICoreWebView2Frame, for example, so it's not actually a frame at the API layer.)

Reply:
There're different frame types, like top-level document (CoreWebView2 directly rendered) is a main frame type. But I agree, will update to use our public API terminology.

Reply:
Yes please update to use only public terminology.